src/adapter.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const { Connection } = require('interface-connection')
```

Comment:
Should we do the async/await update to `interface-connection` for this? It should be a pretty quick change and will prevent us from needing to come back and update this later.

Reply:
@dirkmc do you want to send a PR for this?

cc @vasco-santos (I know you were thinking of working on it this week)

Reply:
I'm not sure what fields / methods should be required on the interface, I opened an issue to discuss: https://github.com/libp2p/interface-connection/issues/28